packages/hedera/src/dids/HederaDidRegistrar.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  AgentContext,
+  DidCreateResult,
+  DidDeactivateResult,
+  DidDocument,
+  DidDocumentKey,
+  DidDocumentRole,
+  DidDocumentService,
+  DidRecord,
+  DidRegistrar,
+  DidRepository,
+  DidUpdateResult,
+  JsonTransformer,
+} from '@credo-ts/core'
+import {
+  HederaDidCreateOptions,
+  HederaDidDeactivateOptions,
+  HederaDidUpdateOptions,
+  HederaLedgerService,
+} from '../ledger/HederaLedgerService'
+
+export class HederaDidRegistrar implements DidRegistrar {
+  public readonly supportedMethods = ['hedera']
+
+  async create(agentContext: AgentContext, options: HederaDidCreateOptions): Promise<DidCreateResult> {
+    try {
+      const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+      const ledgerService = agentContext.dependencyManager.resolve(HederaLedgerService)
+
+      const { did, didDocument, rootKey } = await ledgerService.createDid(agentContext, options)
+
+      const credoDidDocument = new DidDocument({
+        ...didDocument,
+        service: didDocument.service?.map((s) => new DidDocumentService(s)),
+      })
```

Comment:
Usually we use DidDocument.fromJSON() if we transform from a json document, would that work here?